diff --git a/ggml/src/ggml-hexagon/ggml-hexagon.cpp b/ggml/src/ggml-hexagon/ggml-hexagon.cpp
index 48ded82..6f97edb 100644
--- a/ggml/src/ggml-hexagon/ggml-hexagon.cpp
+++ b/ggml/src/ggml-hexagon/ggml-hexagon.cpp
@@ -2230,7 +2230,7 @@ void ggml_hexagon_session::allocate(int dev_id) noexcept(false) {
     this->valid_queue   = false;
     this->valid_iface   = false;
 
-    this->domain_id  = 3;  // Default for CDSP, updated after the session is created
+    this->domain_id  = 4;  // Default for CDSP, updated after the session is created
     this->session_id = 0;  // Default for CDSP, updated after the session is created
     this->dev_id     = dev_id;
     this->name       = std::string("HTP") + std::to_string(dev_id);
@@ -2248,8 +2248,8 @@ void ggml_hexagon_session::allocate(int dev_id) noexcept(false) {
     // Create new session
     if (dev_id != 0) {
         struct remote_rpc_reserve_new_session n;
-        n.domain_name_len  = strlen(CDSP_DOMAIN_NAME);
-        n.domain_name      = const_cast<char *>(CDSP_DOMAIN_NAME);
+        n.domain_name_len  = strlen(CDSP1_DOMAIN_NAME);
+        n.domain_name      = const_cast<char *>(CDSP1_DOMAIN_NAME);
         n.session_name     = const_cast<char *>(this->name.c_str());
         n.session_name_len = this->name.size();
 
@@ -2274,8 +2274,8 @@ void ggml_hexagon_session::allocate(int dev_id) noexcept(false) {
 
         struct remote_rpc_get_uri u = {};
         u.session_id      = this->session_id;
-        u.domain_name     = const_cast<char *>(CDSP_DOMAIN_NAME);
-        u.domain_name_len = strlen(CDSP_DOMAIN_NAME);
+        u.domain_name     = const_cast<char *>(CDSP1_DOMAIN_NAME);
+        u.domain_name_len = strlen(CDSP1_DOMAIN_NAME);
         u.module_uri      = const_cast<char *>(htp_uri);
         u.module_uri_len  = strlen(htp_uri);
         u.uri             = session_uri;
@@ -3888,7 +3888,7 @@ static void ggml_hexagon_init(ggml_backend_reg * reg) {
 
     // Init Arch first since it affects other defaults
     if (!str_arch) {
-        int err = get_hex_arch_ver(CDSP_DOMAIN_ID, &opt_arch);
+        int err = get_hex_arch_ver(CDSP1_DOMAIN_ID, &opt_arch);
         if (err != 0) {
             GGML_LOG_ERROR("ggml-hex: failed to query HTP version (err %d) defaulting to v73\n", err);
             opt_arch = 73;
